core/src/main/java/com/linecorp/armeria/client/Endpoint.java

```diff
@@ -377,13 +384,19 @@ public Endpoint withHost(String host) {
 
         final String normalizedIpAddr = IpAddrUtil.normalize(host);
         if (normalizedIpAddr != null) {
-            return new Endpoint(Type.IP_ONLY, normalizedIpAddr, normalizedIpAddr, port, weight, attributes);
+            return new Endpoint(Type.IP_ONLY, normalizedIpAddr, normalizedIpAddr, port,
+                                weight, attributes, false);
         } else if (isDomainSocketAuthority(host)) {
             return new Endpoint(Type.DOMAIN_SOCKET, host, DOMAIN_SOCKET_IP, DOMAIN_SOCKET_PORT,
-                                weight, attributes);
+                                weight, attributes, false);
         } else {
+            boolean hasTrailingDot = false;
+            if (host.endsWith(".")) {
+                hasTrailingDot = true;
+            }
```

Comment:
```suggestion
            final boolean hasTrailingDot = host.endsWith(".");
```
.. or you could even just inline it because it's used only once.